### PR TITLE
test_diag_manager2.sh Fix

### PR DESCRIPTION
**Pull request by colingladueNOAA**
**Description**
Remedies the skipped tests in test_diag_manager2.sh. 
Specifically:
"Test 12: Catch duplicate field in diag_table"
"Test 19: Area and Volume same field"

Fixes # (issue)
#334 

**How Has This Been Tested?**
Skylake50
make distclean
autoreconf -i
./configure FCFLAGS="-fno-alias -auto -safe-cray-ptr -ftz -assume byterecl -i4 -r8 -nowarn -sox -traceback -qopenmp `nf-config --fflags` -O0 " CPPFLAGS="`nc-config --cflags` -I/opt/intel/2019_up2/impi/2019.2.187/intel64/include/" FC="mpiifort" CC="mpiicc"
make check

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [ ] `make distcheck` passes

### CODE REVIEW

**Comment by uramirez8707** [comment]
@colingladueNOAA There is probably a better way to do this (@underwoo?), but add this to the end of the .travis.yml so we can see the test-suite.log and see why the test is failing here: 

```
after_failure:
  - cat /home/travis/build/NOAA-GFDL/FMS/FMS-2020.02-dev/_build/sub/test_fms/diag_manager/test-suite.log
```

**Comment by underwoo** [comment]
> @colingladueNOAA There is probably a better way to do this (@underwoo?), but add this to the end of the .travis.yml so we can see the test-suite.log and see why the test is failing here:
> 
> ```
> after_failure:
>   - cat /home/travis/build/NOAA-GFDL/FMS/FMS-2020.02-dev/_build/sub/test_fms/diag_manager/test-suite.log
> ```

There is a better way, but this is fine for quick debugging.  Just be sure to remove this when you are ready to have this merged.

Note, the better way is to upload a build artifact containing all the `*.log` files.  Unfortunately, Travis only supports (natively) using Amazon S3.  We could write our own scripts that can upload to the GFDL FTP.

**Comment by thomas-robinson** [comment]
@colingladueNOAA you shouldn't call `stop` to stop an MPI program.  If test 12 doesn't need to do anything until the end, then you can wrap everything in the test 12 if statement, and move the endif just before https://github.com/colingladueNOAA/FMS/blob/master/test_fms/diag_manager/test_diag_manager.F90#L998

File: test_fms/diag_manager/test_diag_manager.F90 [hunk 1]
```diff
@@ -429,7 +429,9 @@ PROGRAM test
      CALL diag_manager_init(err_msg=err_msg)
      IF ( err_msg /= '' ) THEN
         WRITE (out_unit,'(a)') 'test12 successful: err_msg='//TRIM(err_msg)
-        CALL error_mesg('test_diag_manager','test12 successful.',FATAL)
+        CALL error_mesg('test_diag_manager','test12 successful.',NOTE)
+        CALL diag_manager_end(TIME)
+        STOP
```
Comment (thomas-robinson) [inline test_fms/diag_manager/test_diag_manager.F90]: This is your problem.  Calling `stop` on an MPI program in bad.

File: .travis.yml [hunk 2]
```diff
@@ -38,3 +38,7 @@ script:
   - autoreconf -i
   - ./configure ${DISTCHECK_CONFIGURE_FLAGS}
   - make -j distcheck
+  
+after_failure:
+  - cat /home/travis/build/NOAA-GFDL/FMS/FMS-2020.02-dev/_build/sub/test_fms/diag_manager/test-suite.log
+
```
Comment (thomas-robinson) [inline .travis.yml]: Remove this before final PR

**Pull request merged**